troubleshooting/cms_barebones_image_boot.md

```diff
@@ -0,0 +1,253 @@
+- [3. Controlling Output](#csm-boot-output-level)
+- [4. Manual Steps](#csm-boot-manual-steps)
+  - [4.1 Locate CSM Barebones Image in IMS](#csm-boot-steps-locate-barebones-image-in-ims)
+  - [4.2 Create a BOS Session Template for the CSM Barebones Image](#csm-boot-steps-bos-session-template)
+  - [4.3 Find an available compute node](#csm-boot-steps-node)
+  - [4.4 Reboot the node using a BOS session template](#csm-boot-steps-reboot)
+  - [4.5 Connect to the node's console and watch the boot](#csm-boot-steps-watch-boot)
+
+<a name="csm-boot-script-steps"></a>
+## 1. Steps the Script Performs
+
+This script automates the following steps.  If the script fails investigate the underlying service
+to insure it is operating correctly and examine the detailed log file to find information on the
+exact error and cause of failure.
+
+- Obtain the kubernetes API gateway access token
+- Find the existing barebones boot image using IMS
+- Create a BOS session template for the barebones boot image
+- Find an enabled compute node using HSM
+- Watch the console log for the target compute node using console services
+- Create a BOS session to reboot the target compute node
+- Wait for the console output to show an error or successfully reach dracut
+
+The boot may take up to 10 or 15 minutes. The image being booted does not support a complete boot,
+so the node will not boot fully into an operating system. This test is merely to verify that the
+CSM services needed to boot a node are available and working properly. This boot test is considered
+successful if the boot reaches the dracut stage. 
+
+<a name="csm-boot-compute-node"></a>
+## 2. Controlling Which Node Is Used
+
+By default the script will gather all enabled compute nodes that are present in HSM and
+choose one at random to perform the boot test. This may be overridden with a command line
+option to choose which compute node is rebooted using the '--xname' option. The input
+compute node must be enabled and present in HSM to be used. If the input compute node is
+not available a warning will be issued and the test will continue with a valid compute node
+instead of the user selected node.
+
+```bash
+ncn# /opt/cray/tests/integration/csm/barebonesImageTest --xname x3000c0s10b1n0
+```
+
+<a name="csm-boot-output-level"></a>
+## 3. Controlling Test Script Output Level
+
+Output is directed to both the console calling the script as well as a log file that will hold
+more detailed information on the run and any potential problems found.  The log file is written
+to '/tmp/cray.barebones-boot-test.log' and will overwrite any existing file at that location on
+each new run of the script.  
+
+The messages output to the console and the log file may be controlled separately through
+environment variables.  To control the information being sent to the console, set the variable
+'CONSOLE_LOG_LEVEL'.  To control the information being sent to the log file, set the variable
```

Comment:
All of the variable names in this paragraph should be surrounded by backticks instead of single quotes